LayoutTests/resources/accessibility-helper.js

```diff
@@ -266,6 +266,29 @@ async function waitForFocus(id) {
     return focusedElement;
 }
 
+// Selects text within the element with the given ID, and returns the global selected `TextMarkerRange` after doing so.
+async function selectElementTextById(id) {
+    const webArea = accessibilityController.rootElement.childAtIndex(0);
+    const selection = window.getSelection();
+    selection.removeAllRanges();
+
+    await waitFor(() => {
+        const selectedRange = webArea.selectedTextMarkerRange();
+        return webArea.textMarkerRangeLength(selectedRange) == 0;
+    });
+
+    const range = document.createRange();
+    range.selectNodeContents(document.getElementById(id));
+    selection.addRange(range);
+
+    let selectedRange;
+    await waitFor(() => {
+        selectedRange = webArea.selectedTextMarkerRange();
+        return webArea.textMarkerRangeLength(selectedRange) > 0;
+    });
+    return selectedRange;
+}
```

Comment:
Can you elaborate on what is duplicated? This function operates in a few steps:

 1. Clear DOM selection entirely.
 2. Wait for the accessibility tree to report there is no selected text marker range.
 3. Select the text of the element by `id`.
 4. Wait for the accessibility tree to report there is a selected text marker range (its length is greater than zero). Because we cleared the selection in steps 1 and 2, we know the range we get here is of the text we meant to select.
 
If you have a better idea on doing this, let me know.

Reply:
Facepalm, Andres meant that this function is in `accessibility-helper.js` and in the test! Will fix this, thanks.